app/test/vesting_test.go

```diff
@@ -0,0 +1,674 @@
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) balance
+	// to be on the safe side we select one that starts unlocking in at least 20 seconds
+	_, name, err := s.getAnUnusedContinuousVestingAccount(tmtime.Now().Unix() + 20)
+	assert.NoError(s.T(), err)
+
+	s.testDelegatingVestingAmount(name)
+	s.testClaimDelegationReward(name)
+}
+
+// testTransferVestingAmount tests the transfer of vesting amounts (locked balance) from a vesting account
+// to another account. It takes the name of the vesting account (name) as an input.
+// It retrieves a random unused regular account and attempts to transfer the locked amount from the vesting
+// account to the random account. It asserts that the result code of the transaction is equal to 5,
+// indicating a failure in the transfer.
+func (s *VestingModuleTestSuite) testTransferVestingAmount(name string) {
+	randomAcc, err := s.unusedAccount(RegularAccountType)
+	assert.NoError(s.T(), err)
+
+	msgSend := banktypes.NewMsgSend(
+		getAddress(name, s.cctx.Keyring),
+		getAddress(randomAcc, s.cctx.Keyring),
+		sdk.NewCoins(sdk.NewCoin(app.BondDenom, sdk.NewInt(vestingAmount))), // try to transfer the locked amount
+	)
+	resTx, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, name, []sdk.Msg{msgSend}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := s.cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+
+	assert.EqualValues(s.T(), 5, resQ.TxResult.Code, "the transfer TX must fail")
+}
+
+// testDelegatingVestingAmount tests the delegation of vesting amounts (locked) from a vesting account to a validator.
+// It takes the name of the vesting account (name) as an input and attempts to delegate the entire vesting amount to
+// a validator. The delegation transaction should go through and then it retrieves the account delegations again for
+// the given vesting account. It asserts that the delegated amount matches the locked amount.
+func (s *VestingModuleTestSuite) testDelegatingVestingAmount(name string) {
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	del, err := testfactory.GetAccountDelegations(s.cctx.GRPCClient, address)
+	assert.NoError(s.T(), err)
+	assert.Empty(s.T(), del, "initial delegation must be empty")
+
+	validators, err := testfactory.GetValidators(s.cctx.GRPCClient)
+	assert.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), validators)
+
+	msgDelg := stakingtypes.NewMsgDelegate(
+		getAddress(name, s.cctx.Keyring),
+		validators[0].GetOperator(),
+		sdk.NewCoin(app.BondDenom, sdk.NewInt(vestingAmount)),
+	)
+	resTx, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, name, []sdk.Msg{msgDelg}...)
+	assert.NoError(s.T(), err)
+
+	resQ, err := s.cctx.WaitForTx(resTx.TxHash, 10)
+	assert.NoError(s.T(), err)
+	assert.EqualValues(s.T(), 0, resQ.TxResult.Code, fmt.Sprintf("the delegation TX must succeed: \n%s", resQ.TxResult.String()))
```

Comment:
[nit] I think it's clearer to use the abci status codes

```suggestion
	assert.EqualValues(s.T(), abci.CodeTypeOK, resQ.TxResult.Code, fmt.Sprintf("the delegation TX must succeed: \n%s", resQ.TxResult.String()))
```